lib/api/bucketDeleteWebsite.js

```diff
@@ -12,9 +13,10 @@ export default function bucketDeleteWebsite(authInfo, request, log, callback) {
     const canonicalID = authInfo.getCanonicalID();
 
     return metadata.getBucket(bucketName, log, (err, bucket) => {
+        const corsHeaders = collectCorsHeaders(request, bucket);
         if (err) {
             log.debug('metadata getbucket failed', { error: err });
-            return callback(err);
+            return callback(err, corsHeaders);
         }
         if (bucketShield(bucket, requestType)) {
             return callback(errors.NoSuchBucket);
```

Comment:
BucketShield is used to shield buckets from requests when they've been flagged for deletion or 'transient' (in process of creation). In those scenarios we return a `NoSuchBucket` error to the user so I think it's a little odd to return cors headers for a bucket that isn't supposed to exist from the user's perspective. But we technically could, so I'm open to other arguments.